src/qml/components/blockclockdial.cpp

```diff
@@ -210,16 +216,50 @@ void BlockClockDial::paintBlocks(QPainter * painter)
         }
 
         QPainterPath path;
-        path.arcMoveTo(bounds, startAngle);
+        if (i == numberOfBlocks - 1) {
+            // the last block is rounded so nudge it forward
+            // to prevent it from bleeding into the previous
+            path.arcMoveTo(bounds, startAngle - 2 * gap);
+        } else {
+            path.arcMoveTo(bounds, startAngle);
+        }
+
+        if (-1 * nextAngle + 90 > m_animating_max_angle) {
+            nextAngle = -1 * m_animating_max_angle + 90;
+            // end the loop early
+            i = numberOfBlocks;
+        }
+
+        const qreal spanAngle = -1 * (startAngle - nextAngle);
+        if (i == numberOfBlocks - 1) {
+            // nudge the last block foward
+            path.arcTo(bounds, startAngle - 2 * gap, spanAngle);
+        } else {
+            path.arcTo(bounds, startAngle, spanAngle);
+        }
+        painter->drawPath(path);
+    }
+
+    QPen gapPen(m_background_color);
+    gapPen.setWidthF(4);
+    gapPen.setCapStyle(Qt::FlatCap);
+    const QRectF gapBounds = getBoundsForPen(gapPen);
+    painter->setPen(gapPen);
+
+    // Paint Gaps
+    for (int i = 1; i < numberOfBlocks - 1; i++) {
+        qreal nextAngle = 90 + (-360 * m_time_ratio_list[i+1].toDouble());
+
+        QPainterPath path;
+        path.arcMoveTo(gapBounds, nextAngle + gap);
 
         if (-1 * nextAngle + 90 > m_animating_max_angle) {
             nextAngle = -1 * m_animating_max_angle + 90;
             // end the loop early
             i = numberOfBlocks;
         }
 
-        const qreal spanAngle = -1 * (startAngle - nextAngle) + gap;
-        path.arcTo(bounds, startAngle, spanAngle);
+        path.arcTo(gapBounds, nextAngle, gap * 2);
```

Comment:
As a start, this makes the gaps close to what we want them to be

```suggestion
        path.arcTo(gapBounds, nextAngle, gap);
```